test/Microsoft.Health.Dicom.Tests.Integration/Persistence/WorkitemTests.cs

```diff
@@ -38,9 +38,43 @@ public async Task WhenValidWorkitemIsCreated_CreationSucceeds()
                 new QueryTag(new ExtendedQueryTagStoreEntry(2, tag2.GetPath(), tag2.GetDefaultVR().Code, null, QueryTagLevel.Instance, ExtendedQueryTagStatus.Ready, QueryStatus.Enabled,0)),
             };
 
-            long workitemKey = await _fixture.WorkitemStore.AddWorkitemAsync(DefaultPartition.Key, dataset, queryTags, CancellationToken.None);
+            long workitemKey = await _fixture.IndexWorkitemStore.AddWorkitemAsync(DefaultPartition.Key, dataset, queryTags, CancellationToken.None);
 
-            Assert.True(workitemKey > 0, "WorkitemKey not returned.");
+            Assert.True(workitemKey > 0);
+        }
+
+        [Fact]
+        public async Task WhenValidWorkitemIsDeleted_DeletionSucceeds()
+        {
+            string workitemUid = DicomUID.Generate().UID;
+            DicomTag tag2 = DicomTag.PatientName;
+
+            var dataset = new DicomDataset();
+            dataset.Add(DicomTag.SOPInstanceUID, workitemUid);
+            dataset.Add(DicomTag.PatientName, "Foo");
+
+            var queryTags = new List<QueryTag>()
+            {
+                new QueryTag(new ExtendedQueryTagStoreEntry(2, tag2.GetPath(), tag2.GetDefaultVR().Code, null, QueryTagLevel.Instance, ExtendedQueryTagStatus.Ready, QueryStatus.Enabled,0)),
+            };
+
+            long workitemKey = await _fixture.IndexWorkitemStore.AddWorkitemAsync(DefaultPartition.Key, dataset, queryTags, CancellationToken.None);
+
+            await _fixture.IndexWorkitemStore.DeleteWorkitemAsync(DefaultPartition.Key, workitemUid, CancellationToken.None);
+
+            // Try adding it back again, if this succeeds, then assume that Delete operation has succeeded.
```

Comment:
can't we get to check it's not there?

Reply:
Currently we didnt implement GetWorkitem, so this is a temp code, once we have getworkitem done, we can remove it. I will have add a comment